### PR TITLE
Integrated 960 finish integratedcommentb

### PR DESCRIPTION
**Pull request by johanliefers**

### CODE REVIEW

**Comment by integratedforpublishers** [comment]
Can one of the admins verify this patch?

File: EventListener/CommentFormFieldsSubscriber.php [hunk 1]
```diff
@@ -57,112 +79,29 @@ public function onBuildField(FieldEvent $event)
...
-    private function getInnerHtml(\DOMElement $domElement)
+    public function postBuildField(BuilderEvent $event)
     {
-        $innerHTML = "";
+        $field = $event->getBuilder()->get($event->getField());
 
-        /** @var \DOMNode $child */
-        foreach ($domElement->childNodes as $child) {
-            $innerHTML .= $domElement->ownerDocument->saveHTML($child);
+        if ($field->getType()->getName() == 'integrated_editor') {
```
Comment (johnnyborg) [inline EventListener/CommentFormFieldsSubscriber.php]: A code standard in Symfony2 is to use Yoda statements. (see http://symfony.com/doc/current/contributing/code/standards.html#structure)

File: Resources/views/Comment/get.html.twig [hunk 2]
```diff
@@ -1,76 +1,17 @@
-<div class="comment-holder">
+<div class="comment-holder" style="margin-top: 26px;">
```
Comment (johnnyborg) [inline Resources/views/Comment/get.html.twig]: Can you remove the inline styling?

File: Resources/views/Comment/new.html.twig [hunk 3]
```diff
@@ -1,24 +1,22 @@
-<div class="comment-holder hold">
-    <div class="comment-inner">
+<div class="comment-holder hold"  style="margin-top: 26px;">
```
Comment (johnnyborg) [inline Resources/views/Comment/new.html.twig]: Can you remove the inline styling?

File: Util/StripTagsUtil.php [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    }
+
+    /**
+     * Searches for comment tag and replaces it with span tag
+     * @param $content
+     * @return string
+     */
+    public static function replaceCommentWithSpan($content)
+    {
+        $replacement = '<span class="integrated-comment" data-comment-id="$2">$3</span>';
```
Comment (johnnyborg) [inline Util/StripTagsUtil.php]: Or just hardcode the string , or make it a CONST.

File: Util/StripTagsUtil.php [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+class StripTagsUtil
+{
+    /**
+     * Searches for integrated comment span tag and replaces it with a html comment tag
+     * @param $content
+     * @return string
+     */
+    public static function replaceSpanWithCommentTag($content)
+    {
+        $replacement = '<!--integrated-comment=$2-->$3<!--end-integrated-comment-->';
```
Comment (johnnyborg) [inline Util/StripTagsUtil.php]: Or just hardcode the string , or make it a CONST.

File: Twig/Extension/CommentExtension.php [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+     * {@inheritdoc}
+     */
+    public function getFilters()
+    {
+        return [
+            new \Twig_SimpleFilter('remove_comments', [$this, 'escape'], ['is_safe' => ['html']]),
+        ];
+    }
+
+    public function escape($content)
```
Comment (johnnyborg) [inline Twig/Extension/CommentExtension.php]: phpDoc

File: Form/Type/CommentType.php [hunk 7]
```diff
@@ -1,13 +1,22 @@
...
+
 namespace Integrated\Bundle\CommentBundle\Form\Type;
 
 use Symfony\Component\Form\AbstractType;
 use Symfony\Component\Form\FormBuilderInterface;
-use Symfony\Component\OptionsResolver\OptionsResolver;
 
 /**
  * Class CommentType
+ * @package Integrated\Bundle\CommentBundle\Form\Type
```
Comment (johnnyborg) [inline Form/Type/CommentType.php]: I dont believe this is the class header we're using.

File: Form/DataTransformer/CommentTagTransformer.php [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+    }
+
+    /**
+     * Transforms span elements to html comments
+     *
+     * @param  string|null $content
+     * @return string|null
+     * @throws TransformationFailedException
+     */
+    public function reverseTransform($content)
```
Comment (johnnyborg) [inline Form/DataTransformer/CommentTagTransformer.php:57]: Isnt this an {@inheritdoc} ?

File: EventListener/CommentFormFieldsSubscriber.php [hunk 9]
```diff
@@ -1,18 +1,27 @@
...
 use Integrated\Common\Content\Form\Event\BuilderEvent;
 use Integrated\Common\Content\Form\Event\FieldEvent;
 use Integrated\Common\Content\Form\Events;
+
 use Symfony\Component\EventDispatcher\EventSubscriberInterface;
-use Symfony\Component\Form\FormEvent;
-use Symfony\Component\Form\FormEvents;
 
 /**
  * Class ContentCommentSubscriber
```
Comment (johnnyborg) [inline EventListener/CommentFormFieldsSubscriber.php]: Where's the author tag?

File: EventListener/CommentFormFieldsSubscriber.php [hunk 10]
```diff
@@ -25,23 +34,36 @@ class CommentFormFieldsSubscriber implements EventSubscriberInterface
...
-    public function __construct(DocumentManager $documentManager)
+    public function __construct(DocumentManager $documentManager, AssetManager $stylesheets, AssetManager $javascripts)
     {
         $this->documentManager = $documentManager;
-
+        $this->stylesheets = $stylesheets;
+        $this->javascripts = $javascripts;
     }
 
+
```
Comment (johnnyborg) [inline EventListener/CommentFormFieldsSubscriber.php]: PSR2 failure

File: Resources/views/Comment/partial.comment.html.twig [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+            <h4 class="media-heading">{{ comment.author }}</h4>
+        {% endif %}
+        <p>
+            <small>
+                <span class="date">
+                    {{ comment.date|date('d-m-Y H:s') }}
+                </span>
+            </small>
+            -
+            <a class="delete-comment" href="
```
Comment (johnnyborg) [inline Resources/views/Comment/partial.comment.html.twig:21]: Can you remove the delete? A comment on the issue stated "Next to the code review, also please comment the "Edit", "Delete" and "Resolved" as they have not been implemented yet."

Comment (johanliefers) [reply Resources/views/Comment/partial.comment.html.twig:21]: The buttons don't look nice, no need to show them as comment, when it needs to be implemented someone has to see how they must look like.

**Pull request merged**